app/models/work_group.rb

```diff
@@ -71,6 +71,14 @@ def minified_url(user)
     end
   end
 
+  def assign_colors()
+    # orange, yellow, green, blue, pink, purple
+    colors = ['#FF5733', '#FFE632', '#11841F', '#2B86E7', '#E72BD8', '#B554FF']
+    users.each_with_index do |student, index|
+      student.update(pen_colour: colors[index])
```

Comment:
I chose to do this rather than randomly selecting a color from the array because it seemed easier, but if the students are always added to a group in the same order I made need to think of another way (unless it's fine for a student to have the same color in a few work groups)

Reply:
you can do `array.take(4)` - it will randomly take 4 items from said array that you can then go and assign